src/transformers/file_utils.py

```diff
@@ -239,14 +253,47 @@ def cached_path(
         )
     elif os.path.exists(url_or_filename):
         # File, and it exists.
-        return url_or_filename
+        output_path = url_or_filename
     elif urlparse(url_or_filename).scheme == "":
         # File, but it doesn't exist.
         raise EnvironmentError("file {} not found".format(url_or_filename))
     else:
         # Something unknown
         raise ValueError("unable to parse {} as a URL or as a local path".format(url_or_filename))
 
+    if extract_compressed_file:
+        if not is_zipfile(output_path) and not tarfile.is_tarfile(output_path):
+            return output_path
+
+        # Path where we extract compressed archives
+        # We avoid '.' in dir name and add "-extracted" at the end: "./model.zip" => "./model-zip-extracted/"
+        output_dir, output_file = os.path.split(output_path)
+        output_extract_dir_name = output_file.replace(".", "-") + "-extracted"
+        output_path_extracted = os.path.join(output_dir, output_extract_dir_name)
+
+        if os.path.isdir(output_path_extracted) and os.listdir(output_path_extracted) and not force_extract:
```

Comment:
nit: the `os.listdir()` call here checks that the dir is not empty, and relies on the fact that in Python, `bool([]) == False`, which to me is always a surprise 

Sometimes I'm writing an explicit `len(os.listdir) == 0`

Reply:
I agree but I have a linter warning that using `len(list) == 0` is not a good practice.
There are lengthy discussions on this e.g. here: https://stackoverflow.com/a/45778282
In general, I like the readability of `len(os.listdir) == 0` better.

Reply:
interesting read, thanks